packages/transactions/src/__typetests__/transactions-typetests.ts

```diff
@@ -0,0 +1,408 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+import { Ed25519Signature } from '@solana/keys';
+
+import {
+    appendTransactionInstruction,
+    Blockhash,
+    Nonce,
+    prependTransactionInstruction,
+    setTransactionLifetimeUsingBlockhash,
+    setTransactionLifetimeUsingDurableNonce,
+    signTransaction,
+} from '..';
+import { createTransaction } from '../create-transaction';
+import { setTransactionFeePayer } from '../fee-payer';
+import { compileMessage } from '../message';
+import { Transaction } from '../types';
+import { getUnsignedTransaction } from '../unsigned-transaction';
+
+const mockFeePayer = 'feePayer' as Base58EncodedAddress<'feePayer'>;
+const mockBlockhash = 'blockhash' as Blockhash;
+const mockBlockhashLifetime = {
+    blockhash: mockBlockhash,
+    lastValidBlockHeight: 0n,
+};
+const mockSigner = {} as CryptoKeyPair;
+const mockNonceConfig = {
+    nonce: 'nonce' as Nonce<'nonce'>,
+    nonceAccountAddress: 'nonce' as Base58EncodedAddress<'nonce'>,
+    nonceAuthorityAddress: 'nonceAuthority' as Base58EncodedAddress<'nonceAuthority'>,
+};
+const mockInstruction = {
+    accounts: [],
+    data: Uint8Array.of(0),
+    programAddress: 'program' as Base58EncodedAddress<'program'>,
+} as Transaction['instructions'][number];
+
+function assertNotAProperty<T extends object, TPropName extends string>(
+    _: { [Prop in keyof T]: Prop extends TPropName ? never : T[Prop] },
+    _propName: TPropName
+): void {}
+
+async () => {
+    // createTransaction
+    {
+        const mockLegacyTransaction = createTransaction({ version: 'legacy' });
+        mockLegacyTransaction satisfies Transaction & { version: 'legacy' };
```

Comment:
I think this is more like ‘satisfies the member of the union whose version is …’


```suggestion
        mockLegacyTransaction satisfies Extract<Transaction, { version: 'legacy' }>;
```